pom.xml

```diff
@@ -39,157 +58,131 @@
+  <dependencyManagement>
+    <dependencies>
+
+      <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-changelog</artifactId>
+        <version>${project.version}</version>
+      </dependency>
+
+      <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-docgen</artifactId>
+        <version>${project.version}</version>
+      </dependency>
+
+      <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-docgen-asciidoctor-extension</artifactId>
+        <version>${project.version}</version>
+      </dependency>
+
+    </dependencies>
+  </dependencyManagement>
+
   <build>
     <plugins>
+
+      <!-- Enable BOM flattening -->
       <plugin>
-        <groupId>org.apache.rat</groupId>
-        <artifactId>apache-rat-plugin</artifactId>
-        <version>0.11</version>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>flatten-maven-plugin</artifactId>
+        <version>${flatten-maven-plugin.version}</version>
+        <executions>
+          <execution>
+            <id>flatten-bom</id>
+            <goals>
+              <goal>flatten</goal>
+            </goals>
+            <phase>process-resources</phase>
+            <inherited>false</inherited>
+          </execution>
+        </executions>
       </plugin>
+
+      <!-- *Temporarily* override the changelog entry file validation inherited from `logging-parent`.
+
+           Inherited `validate-changelog` execution explicitly fixes the `log4j-changelog` namespace and schema location.
+           This makes it impossible to upgrade changelog entry file schemas to a newer version than the one expected by `logging-parent`.
+
+           TODO: This will probably be fixed in `logging-parent` version `11.0.0` release.
+                 Hence, from then on, this `xml-maven-plugin` override can be removed. -->
       <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-site-plugin</artifactId>
-        <version>3.4</version>
-        <configuration>
-          <templateFile>${basedir}/src/site/site.vm</templateFile>
-        </configuration>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>xml-maven-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>validate-changelog</id>
+            <configuration combine.self="override">
+              <validationSets>
+                <validationSet>
+                  <dir>src/changelog</dir>
+                  <includes>
+                    <include>**/*.xml</include>
+                  </includes>
+                  <validating>true</validating>
+                </validationSet>
+              </validationSets>
+            </configuration>
+          </execution>
+        </executions>
       </plugin>
-    </plugins>
-  </build>
-  <reporting>
-    <plugins>
+
       <plugin>
```

Comment:
@ppkarwasz, do we have any Spotless configuration this project can take advantage of? Otherwise, shall @FreeAndNil remove this plugin execution?

Reply:
I think spotless checks for the license headers, but I'm not sure.

Reply:
I've removed Spotless.